### PR TITLE
Added support for dirty-checking collections

### LINKED ISSUE #26: Updating Custom Fields?
Custom Fields don't seem to be sticking when trying to assign them. Perhaps its something I'm doing wrong. 

I've tried using GetLoanAsync and updating the CF in the loan.CustomFields object as well as creating a new loan using the Loan GUID and adding the CF. 

When I use Loans.UpdateLoanAsync The Standard Field (Property State) that I updated has the new data but the Custom Field Still has the Old data in it. Here is the Code I'm Using:

Loan loan = await client.Loans.GetLoanAsync(item.LoanGuid);

                            loan.Property.State = "OH";

                            foreach (CustomField cf in loan.CustomFields)
                            {
                                if (cf.FieldName == "CUST91FV")
                                {
                                    cf.StringValue = "Delete this!";
                                }
                            }

                            await client.Loans.UpdateLoanAsync(loan, true).ConfigureAwait(false);

### PR DESCRIPTION
**Pull request by TylerBrinkley**
Added support for dirty-checking collections. Had to change exposed type to `IList<T>` instead of `List<T>`.

Made entity and collection property getters new up references when null.

Added serialization logic to always serialize the `Id` property of objects except in the case of `CustomField`s which instead always serialize the `FieldName` property.

Fixes #26.

### CODE REVIEW

**Comment by hfcjweinstock** [comment]
do we want to serialize the entire collection for a single change?  am i reading this correctly?

**Comment by TylerBrinkley** [comment]
No, we're not. Checkout the `DirtyListConverter` in the `DirtyList.cs` file.

**Comment by TylerBrinkley** [comment]
@hfcjweinstock any thoughts on this one? I know it's kind of a bear but it's not too bad if you ignore the entities that are changing. The main files that changed are `DirtyList.cs`, `DirtyValue.cs` and `JsonHelper.cs`.

File: src/EncompassRest/DirtyList.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            throw new NotSupportedException();
+        }
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var dirtyList = (DirtyList<T>)value;
+            var listToSerialize = new List<T>();
+            foreach (var item in dirtyList._list)
```
[the remark this answers is not in the document]
Comment (TylerBrinkley) [reply src/EncompassRest/DirtyList.cs]: Yep, just updated it. Don't know why it took so long for your comment to post.

**Pull request merged**